spring-pulsar/src/main/java/org/springframework/pulsar/core/reactive/DefaultReactivePulsarSenderFactory.java

```diff
@@ -0,0 +1,99 @@
+import org.apache.pulsar.reactive.client.adapter.AdaptedReactivePulsarClientFactory;
+import org.apache.pulsar.reactive.client.api.ReactiveMessageSender;
+import org.apache.pulsar.reactive.client.api.ReactiveMessageSenderBuilder;
+import org.apache.pulsar.reactive.client.api.ReactiveMessageSenderSpec;
+import org.apache.pulsar.reactive.client.api.ReactivePulsarClient;
+
+import org.springframework.core.log.LogAccessor;
+import org.springframework.util.CollectionUtils;
+
+/**
+ * Default implementation of {@link ReactivePulsarSenderFactory}.
+ *
+ * @param <T> reactive sender type.
+ * @author Christophe Bornet
+ */
+public class DefaultReactivePulsarSenderFactory<T> implements ReactivePulsarSenderFactory<T> {
+
+	private final LogAccessor logger = new LogAccessor(LogFactory.getLog(this.getClass()));
+
+	private final ReactivePulsarClient reactivePulsarClient;
+
+	private final ReactiveMessageSenderSpec reactiveMessageSenderSpec;
+
+	public DefaultReactivePulsarSenderFactory(PulsarClient pulsarClient,
+			ReactiveMessageSenderSpec reactiveMessageSenderSpec) {
+		this(AdaptedReactivePulsarClientFactory.create(pulsarClient), reactiveMessageSenderSpec);
+	}
+
+	public DefaultReactivePulsarSenderFactory(ReactivePulsarClient reactivePulsarClient,
+			ReactiveMessageSenderSpec reactiveMessageSenderSpec) {
+		this.reactivePulsarClient = reactivePulsarClient;
+		this.reactiveMessageSenderSpec = reactiveMessageSenderSpec;
+	}
+
+	@Override
+	public ReactiveMessageSender<T> createReactiveMessageSender(String topic, Schema<T> schema) {
+		return doCreateReactiveMessageSender(topic, schema, null, null);
+	}
+
+	@Override
+	public ReactiveMessageSender<T> createReactiveMessageSender(String topic, Schema<T> schema,
+			MessageRouter messageRouter) {
+		return doCreateReactiveMessageSender(topic, schema, messageRouter, null);
+	}
+
+	@Override
+	public ReactiveMessageSender<T> createReactiveMessageSender(String topic, Schema<T> schema,
+			MessageRouter messageRouter, List<ReactiveMessageSenderBuilderCustomizer<T>> customizers) {
+		return doCreateReactiveMessageSender(topic, schema, messageRouter, customizers);
+	}
+
+	private ReactiveMessageSender<T> doCreateReactiveMessageSender(String topic, Schema<T> schema,
+			MessageRouter messageRouter, List<ReactiveMessageSenderBuilderCustomizer<T>> customizers) {
+		final String resolvedTopic = ReactiveMessageSenderUtils.resolveTopicName(topic, this);
+		this.logger.trace(() -> String.format("Creating reactive message sender for '%s' topic", resolvedTopic));
+		final ReactiveMessageSenderBuilder<T> sender = this.reactivePulsarClient.messageSender(schema);
+		if (this.reactiveMessageSenderSpec != null) {
+			sender.applySpec(this.reactiveMessageSenderSpec);
+		}
+		sender.topic(resolvedTopic);
+		if (messageRouter != null) {
+			sender.messageRouter(messageRouter);
+		}
+		if (!CollectionUtils.isEmpty(customizers)) {
+			customizers.forEach((c) -> c.customize(sender));
+		}
+		return sender.build();
```

Comment:
One thing I noticed is there are no producer interceptors in the RMS. Has that not been added or is it achieved another way?